### DISCUSSION TITLE
About the ETL from kafka to ravendb, is

### DISCUSSION THREAD

**Discussion by ValiLucifer89** [ravendb/ravendb#15093] [Q&A] [answered]
Kafka clients can be created for at-most-once, at-least-once, and exactly-once message processing needs. From kafka to ravendb, what is the processing like at-most-once, at-least-once, and exactly-once?

**Comment by arekpalinski** [comment, marked as answer]
Note that our ETL is about sending data from RavenDB to Kafka (not the other way around).

https://ravendb.net/docs/article-page/5.4/csharp/server/ongoing-tasks/etl/queue-etl/kafka#data-delivery

The relevant pieces of code where we configure Kafka producer are here:
https://github.com/ravendb/ravendb/blob/6d637682550d83ca56094b44b7596d9d3bbeec6b/src/Raven.Server/Documents/ETL/Providers/Queue/QueueBrokerConnectionHelper.cs#L16-L45

https://github.com/ravendb/ravendb/blob/6d637682550d83ca56094b44b7596d9d3bbeec6b/src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs#L44-L49
